javascripts/2048.js

```diff
@@ -1,32 +1,276 @@
+
+    var new_column = rando_num();
+    var new_row = rando_num();
+
+    var result = checkLocation(new_column, new_row);
+
+    var column = result[0];
+    var row = result[1];
+
+    // Create new div element with tile class, location attributes, and value attribute
+    var newTile = $( '<div= class="tile">2</div>');
+    newTile.attr("data-col", 'c' + column);
+    newTile.attr("data-row", 'r' + row);
+    newTile.attr("data-val", "2");
+    $('#gameboard').append(newTile);
+
+    setTileInArray(column, row, newTile);
+    pop(newTile);
+}
+
+
+function tileCollide(tile, neighbor, direction) {
+  // tile should be checked against neighbor
+  //console.log('tile', tile, 'neighbor', neighbor);
+  if (tile.attr('data-val') == neighbor.attr('data-val')) {
+    //move tile to neighbors spot
+    var new_col = neighbor.attr('data-col');
+    var new_row = neighbor.attr('data-row');
+    tile.attr('data-col', new_col);
+    tile.attr('data-row', new_row);
+    //console.log(tile);
+    // take col and row and number and set for tile to be deleted
+    var col = +(new_col.slice(1));
+    var row = +(new_row.slice(1));
+
+    // reset tile array so updated tile is in correct spot
+    setTileInArray(col, row, tile);
+
+    // update tile
+    updateTile(tile);
+
+    // remove tile
+    setTimeout(function() {
+      deleteTile(neighbor);
+    }, 500);
+
+
+    // reset tile array so deleted spot is set to undefined
+    resetTileArray(col, row, direction)
+  }
+}
+
+  function deleteTile(neighbor){
+    junkTile = neighbor;
+    console.log("junk", junkTile)
+    $(junkTile).remove();
+  }
+
+
+  function resetTileArray(col, row, direction){
+    if (direction == 'up'){
+      var row = (row + 1);
+      setUndefinedTile(col, row);
+    } else if (direction == 'down') {
+      var row = (row - 1);
+      setUndefinedTile(col, row);
+    } else if (direction == 'left') {
+      var col = (col + 1);
+      setUndefinedTile(col, row);
+    } else if (direction == 'right') {
+      var col = (col - 1);
+      setUndefinedTile(col, row);
+    }
+  }
+
+ function updateTile(tile) {
   var new_tile_value = tile.attr("data-val") * 2;
   tile.attr("data-val", new_tile_value);
   tile.text(new_tile_value);
+ }
+
 
+function moveTile(direction) {
+  // needs conditionals to check for occupied grids spaces
   switch(direction) {
     case 38: //up
-      tile.attr("data-row","r0");
+      //tile.attr("data-row","r0");
+      moveDirection(moveUp);
       break;
     case 40: //down
-      tile.attr("data-row","r3");
+
+      moveDirection(moveDown);
       break;
     case 37: //left
-      tile.attr("data-col","c0");
+
+      moveDirection(moveLeft);
       break;
     case 39: //right
-      tile.attr("data-col","c3");
+
+      moveDirection(moveRight);
       break;
   }
 }
+
+// Loops through to move elements in every column/row
+function moveDirection(moveWay) {
+  for (var i=0; i < 4; i++) {
+    moveWay(i);
+  }
+  setTimeout(function() {
```

Comment:
You may not want to set a delay on creating a tile. Unless you lock the state of the gameboard during this specific delay, it's possible to spam the arrow keys and create race conditions where it will try to create the same tile more than once.